### PR TITLE
feat: provide validation report with 400 errors

### PR DESCRIPTION
**Pull request by martinmaillard**
Part of #103 (solution 1)

### CODE REVIEW

File: apis/core/lib/errors/validation.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import { ProblemDocument } from 'http-problem-details'
+import { ValidationReport } from 'rdf-validate-shacl'
+
+export class ValidationError extends Error {
+  readonly validationReport: ValidationReport
+
+  constructor(validationReport: ValidationReport) {
+    super()
+
+    this.message = 'Bad request'
```
Comment (tpluscode) [inline apis/core/lib/errors/validation.ts:11]: Maybe this could be more informative?

File: apis/core/lib/errors/validation.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    this.validationReport = validationReport
+  }
+}
+
+export class ValidationErrorMapper extends ErrorMapper {
+  private readonly status: number;
+
+  public constructor() {
+    super(ValidationError)
+    this.status = 400
```
Comment (tpluscode) [inline apis/core/lib/errors/validation.ts:21]: Is this used for anything else? Might just inline in `mapError` I suppose

**Review by tpluscode — APPROVED** [review]

File: apis/core/lib/errors/validation.ts (hunk 2 above)
Comment (martinmaillard) [reply apis/core/lib/errors/validation.ts:21]: As often, I don't really understand the whole contraption we are creating. So probably not?

**Review by vhf — APPROVED** [review]

**Review by tpluscode — APPROVED** [review]
Let's squash-merge

**Pull request merged**